vllm/model_executor/models/llama.py

```diff
@@ -64,11 +82,22 @@ def __init__(
         bias: bool = False,
     ) -> None:
         super().__init__()
-        self.gate_up_proj = MergedColumnParallelLinear(
-            input_size=hidden_size,
-            output_sizes=[intermediate_size] * 2,
-            bias=bias,
-            quant_config=quant_config)
+        self.merge_weight = getattr(quant_config, "merge_weight", True)
+        if self.merge_weight:
+            self.gate_up_proj = MergedColumnParallelLinear(
+                input_size=hidden_size,
+                output_sizes=[intermediate_size] * 2,
+                bias=bias,
+                quant_config=quant_config)
+        else:
+            self.gate_proj = ColumnParallelLinear(hidden_size,
+                                                  intermediate_size,
+                                                  bias=bias,
+                                                  quant_config=quant_config)
+            self.up_proj = ColumnParallelLinear(hidden_size,
+                                                intermediate_size,
+                                                bias=bias,
+                                                quant_config=quant_config)
```

Comment:
Unfortunately the`merge_weight` condition isn't acceptable and we need to remove this. We need this quantization method to work with `MergedColumnParallelLinear` and `QKVParallelLinear` since all other methods work on this abstraction. In the worst case where you need to perform these matmuls separately, you should be able to handle this inside of the `GGUFLinearMethod` class during weight loading and/or the forward pass.

Reply:
This case is that imatrix quantization uses different quantization for `q_proj`, `k_proj` and `v_proj` with different block size. 

Besides perform these matmuls separately, we also need to add extra padding to the quantized weights to make sure the weights can be merged with same row_size, which means this will increase VRAM usage (I think this increasement should be acceptable).